src/Lorentz/Contracts/RegistryDAO/Types.hs

```diff
@@ -0,0 +1,162 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+-- | RegistryDAO Types
+module Lorentz.Contracts.RegistryDAO.Types
+  ( RegistryDaoContractExtra (..)
+  , RegistryEntry (..)
+  , RegistryUpdate (..)
+  , RegistryDaoProposalMetadata (..)
+  , NormalProposal (..)
+  , ConfigProposal (..)
+  , IsoRegistryDaoProposalMetadata
+  ) where
+
+import Lorentz
+
+import qualified Lorentz.Contracts.BaseDAO.Types as DAO
+
+
+data RegistryDaoContractExtra k v = RegistryDaoContractExtra
+  { ceRegistry :: BigMap k (RegistryEntry v)
+  , ceFrozenScaleValue :: Natural
+  , ceFrozenExtraValue :: Natural
+  , ceSlashScaleValue :: Natural
+  , ceSlashDivisionValue :: Natural
+  , ceMaxProposalSize :: Natural
+  }
+  deriving stock (Generic)
+
+instance (NiceComparable k, Ord k, IsoValue k, IsoValue v)
+  => (IsoValue (RegistryDaoContractExtra k v))
+
+instance (HasAnnotation k, HasAnnotation v)
+  => HasAnnotation (RegistryDaoContractExtra k v) where
+    annOptions = DAO.baseDaoAnnOptions
+
+instance (Ord k, TypeHasDoc k, TypeHasDoc v)
+  => TypeHasDoc (RegistryDaoContractExtra k v) where
+    typeDocMdDescription =
+      "TODO"
+    typeDocMdReference = poly2TypeDocMdReference
+    typeDocHaskellRep = concreteTypeDocHaskellRep @(RegistryDaoContractExtra ByteString ByteString)
+    typeDocMichelsonRep = concreteTypeDocMichelsonRep @(RegistryDaoContractExtra ByteString ByteString)
+
+instance Default (RegistryDaoContractExtra k v) where
+  def = RegistryDaoContractExtra
+    { ceRegistry = def
+    , ceFrozenScaleValue = 1
+    , ceFrozenExtraValue = 0
+    , ceSlashScaleValue = 1
+    , ceSlashDivisionValue = 1
+    , ceMaxProposalSize = 100
+    }
+
+instance IsoRegistryDaoProposalMetadata k v =>
+    StoreHasSubmap (RegistryDaoContractExtra k v) "ceRegistry" k (RegistryEntry v) where
+  storeSubmapOps = storeSubmapOpsDeeper #ceRegistry
+
+
+data RegistryEntry v = RegistryEntry
+  { reValue :: Maybe v
+   -- optional statistics on given proposal
+  , reLastAgoraPostId :: Natural
+  , reLastUpdated :: Timestamp
```

Comment:
It doesn't match what's written in https://github.com/tqtezos/baseDAO/blob/master/docs/registry.md. We decided to store `ProposalKey` as this "optional statistics", one can query metadata by this key because we don't remove old proposals.

Reply:
> Alternatively, we can merge them and store `option v` values.
It should be treated as an implementation detail.
There is additional data to implement configuration lambdas, it's an implementation detail as well.

In this case, I used `option v` alternative instead. I believe this serves the same purpose as having 2 maps.

Reply:
Yes, but we don't need to store `reLastAgoraPostId`, we can store `ProposalKey` and one can figure out the post ID by looking up the proposal with given key.

Reply:
I see, I got this now. I'll update this to use `lastProposalKey` instead of `reLastAgoraPostId`.

Reply:
I've updated this to be a proposal key instead. For now, I'll focus on making the contract smaller to fix the CI in https://github.com/tqtezos/baseDAO/pull/71